src/map/alliance.cpp

```diff
@@ -237,6 +254,14 @@ void CAlliance::addParty(CParty* party)
         return;
     }
 
+    for (auto* PChar : party->members)
+    {
+        if (auto& PTreasurePool = static_cast<CCharEntity*>(PChar)->GetTreasurePool(); PTreasurePool.IsManaged())
```

Comment:
I know mobs can be in parties, but I don't know if they can have alliances - is it safe to assume entities in here are players and we can `static_cast` them? This isn't the hot path, so we could be checking their objType